plugin/keyring_vault/CMakeLists.txt

```diff
@@ -1,3 +1,14 @@
+IF (NOT WITH_KEYRING_VAULT)
+  MESSAGE (STATUS "Not building with keyring_vault plugin, you must specify -DWITH_KEYRING_VAULT=1|ON|FORCE to cmake in order to build with keyring_vault.")
```

Comment:
The message suggests that the plugin will not be attempted to be built if the option is not specified?

Reply:
Yes, this is the behavior rocksdb is following. Should it be build by default and disabled when WITH_KEYRING_VAULT=0 ?

Reply:
Yes, I believe it should be attempted to be built by default. Not sure re. WITHOUT_ variable, I wouldn't miss it, but what about others? @percona-ysorokin

Reply:
Guys, I am strictly against any kind of implicit actions in the build process and silently skipping keyring_vault plugin, keyring_vault unit tests or both is one of them. From my own experience during the latest upstream merge, it turned out that everything was "OK" with 'keyring_vault' plugin only because it wasn't built at all on 50% of Jenkins nodes - it was silently skipped there. However, there were actual code errors.

So, my suggestion is to have only 2 options `-DWITH_KEYRING_VAULT=ON | OFF`.
When `OFF` - the plugin and its unit tests are never built.
When 'ON' - will attempt to build both the plugin and its unit tests and will generate an error and stop if some dependencies cannot be resolved or there is an actual compilation/linking error.

I also suggest the default to be `ON`.

Reply:
This could be done for keyring_vault plugin, but not for the unittests. The reason being that unittests are only built when CURL does not depend on RTMP library. In case of unittests it has to be always a warning if they cannot be built. It is not a big problem, as unittests passing on just one jenkins machine is enough (cento6  currently). The rest of the machines run MTR tests.

Reply:
I agree with @percona-ysorokin. This will give us extra Jenkins config pain (this platform should build that, that platform should not build this etc), but it's the best way to ensure we don't silently break non-universally built parts

Reply:
Just an idea: can't we add a WITH_CURL=bundled/system option, similar to the other 3rd parties? As I see, curl is also built with cmake, so this shouldn't be too hard to do.

Reply:
@dutow bundled means built in into PS. For WITH_SSL we use bundled for building with YASSL that is shipped with PS. It is already possible to built with custom CURL by providing its path to WITH_CURL=, without the custom path it will just be built with the system CURL.

Reply:
So, let me summarize what we've agreed on.
A new `-DWITH_KEYRING_VAULT=ON | OFF` will be introduced to Percona Server build options.
The default is `ON`.

When `ON`, we will try to build `keyring_vault` plugin. If there is a missing dependency, Cmake will return a fatal error. We will also try to build 'keyring_vault' unit tests. It should also generate a fatal error on a missing dependency. The only exception is that it may generate a warning and skip building 'keyring_vault' unit tests when it finds a system `curl` library which is build with `RTMP` support.

When `OFF`, neither `keyring_vault` plugin nor `keyring_vault` unit tests are built.